### ISSUE TITLE
Default values in namedtuple not supported

### ISSUE THREAD

**Issue by tjnaughton** [brython-dev/brython#1053]
Hi,

With Python 3.7.0
````
>>> from collections import namedtuple
>>> N = namedtuple('N', 'a b')
>>> N.__new__.__defaults__ = (3, 4)
>>> N()
N(a=3, b=4)
````
however with Brython
````
Brython 3.7.1 on Netscape 5.0 (X11)
>>> from collections import namedtuple
>>> N = namedtuple('N', 'a b')
>>> N.__new__.__defaults__ = (3, 4)
>>> N()
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "<string>", line 1, in __new__
TypeError: __new__() missing 2 positional arguments: a,b
````
Thank you!

**Comment by PierreQuentel** [comment]
The problem is more general, setting `__defaults__` has no effect on functions
```python
def f(x):
    return x + 1

f.__defaults__ = (3,)
print(f())
```
raises the same error instead of printing 4.

I can't say that this feature is overdocumented in Python, and the behaviour is a little stange: 
```
Python 3.7.0 (v3.7.0:1bf9cc5093, Jun 27 2018, 04:59:51) [MSC v.1914 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> def f(x):
...  return x + 1
...
>>> f.__defaults__ = (3,)
>>> f()
4
>>> f.__defaults__ = (3, 4, 99)
>>> f()
100
>>>
```
I would have expected the second setting of `f.__defaults__` to raise an exception because there are more values than parameters to the function.

**Comment by tjnaughton** [comment]
You're right, but at least it seems that the behaviour is predictable: the default values are always applied from right-to-left, e.g:
````
Python 3.7.0 (default, Jun 28 2018, 13:15:42)
Type "copyright", "credits" or "license" for more information.
>> def g(x, y):
..     return x + y
..
>> g.__defaults__ = (3,)
>> g(x=10)
13
````

**Comment by tjnaughton** [comment]
Since I posted the issue, I have learned that my example above is out-of-date and that Python 3.7 introduces a new way to define default arguments in namedtuples (the keyword argument `defaults`). It seems that Brython already supports some of this intended behaviour:
````
Brython 3.7.1 on Netscape 5.0 (X11)
>>> from collections import namedtuple
>>> N = namedtuple('N', 'a b', defaults=(3, 4))
>>> N._fields_defaults
{'a': 3, 'b': 4}
````
I don't know, would it be easier to implement Brython support for this way of achieving default nametuple arguments rather than through ``__new__.__defaults__``? After all, now that the keyword argument `defaults` exists, probably nobody will ever try to use the writable ``__new__.__defaults__`` any more.
